### PR TITLE
add sanitizer code for mlflow reporting

### PR DESCRIPTION
**Pull request by ajslone**
This PR adds sanitizer code for mlflow reporting to conform with that package's restrictions on keys and values for tags, parameters, and metrics. This also adds tests for these changes.

### CODE REVIEW

**Comment by ajslone** [comment]
This should wait for https://github.com/google/uv-metrics/pull/21
This has now been merged.

File: uv/reporter/base.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
```
Comment (ajslone) [inline uv/reporter/base.py:18]: Changes here are for mypy to accomodate our new support for nested dictionaries as params.

File: uv/reporter/base.py [hunk 2]
```diff
@@ -200,11 +200,9 @@ def stepped_reporter(base: AbstractReporter,
   is useful for keeping track of each metric's timestamp.
 
   """
-  if step_key is None:
-    step_key = "step"
```
Comment (ajslone) [inline uv/reporter/base.py:203]: For some reason mypy likes this format, but not the prior one.

File: uv/reporter/base.py [hunk 3]
```diff
@@ -350,7 +348,10 @@ def report(self, step: int, k: t.MetricKey, v: t.Metric) -> None:
     self._base.report(step, newk, v)
```
Comment (ajslone) [inline uv/reporter/base.py:349]: `self._base.reader()` returns `Optional`

**Review by sritchie — APPROVED** [review]
Hey, realized I forgot to submit this review, looks great!

**Pull request merged**